flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -45,9 +49,12 @@ public void reset() {
     }
 
     @Override
-    public void add(RowData value) {
+    public void add(KeyValue kv) {
+        checkArgument(
+                kv.valueKind() == RowKind.INSERT || kv.valueKind() == RowKind.UPDATE_AFTER,
+                "Partial update only accept insert only records.");
```

Comment:
What does `insert only records` mean? IMO, the user may not understand this message.